apps/api/src/app/workflows-v2/mappers/workflow-template-get-mapper.ts

```diff
@@ -0,0 +1,80 @@
+import { IPreferenceChannels, StepTypeEnum } from '@novu/shared';
+import { DiscoverWorkflowOutputPreferences } from '@novu/framework';
+import { NotificationStepEntity, NotificationTemplateEntity } from '@novu/dal';
+
+import { MinifiedResponseWorkflowDto, StepDto, WorkflowResponseDto } from '../dto/workflow.dto';
+
+export class WorkflowTemplateGetMapper {
+  static toResponseWorkflowDto(template: NotificationTemplateEntity): WorkflowResponseDto {
+    return {
+      _id: template._id,
+      workflowId: template.name,
+      tags: template.tags,
+      active: template.active,
+      payload: template.data ? { schema: template.data } : undefined, // Assuming data is mapped to payload schema
+      controls: template.payloadSchema ? { schema: template.payloadSchema } : undefined,
+      critical: template.critical,
+      notificationGroupId: template._notificationGroupId,
+      preferences: WorkflowTemplateGetMapper.toPreferences(template.preferenceSettings), // Assuming this directly maps; may need adjustment
+      code: template.name, // Assuming name is used as code
```

Comment:
not the same thing, at the moment we do not store the workflow execute code, we should consider storing it as well if needed.